client/app/scripts/superdesk-packaging/packaging.js

```diff
@@ -331,9 +331,13 @@
             templateUrl: 'scripts/superdesk-packaging/views/sd-package-items-edit.html',
             link: function(scope, elem, attrs, ngModel) {
                 scope.$on('package:addItems', function(event, data) {
-                    var groupIndex = _.findIndex(scope.list, {id: 'main'});
-                    for (var i = 0; i < data.length; i++) {
-                        scope.list[groupIndex].items.unshift(packages.getReferenceFor(data[i]));
+                    var groupIndex = _.findIndex(scope.list, {id: data.group});
+                    if (groupIndex === -1) {
+                        scope.list.push({id: data.group, items: []});
+                        groupIndex = scope.list.length - 1;
+                    }
+                    for (var i = 0; i < data.items.length; i++) {
+                        scope.list[groupIndex].items.unshift(packages.getReferenceFor(data.items[i]));
                     }
```

Comment:
Readability counts and helps a lot with the general maintainability of the code base, thus it should be made a higher priority IMO.

Optimizations that hinder readability should generally only be used if profiling shows that a particular piece of code indeed represents a bottleneck.